### PR TITLE
Space Traveler

### PR DESCRIPTION
**Pull request by Roland-Ntwali**
## In this PR, we did the following:

- Added React Redux, Redux Logger, and React Router dependencies.
- Implemented components, routes, and Redux state management for rockets, missions, and dragons (if applicable).
- Fetched data from SpaceX API and rendered UI based on requirements.
- Created actions and reducers for booking, joining, canceling, and leaving rockets, dragons, and missions.
- Developed My Profile section to display reserved rockets, joined missions, and reserved dragons.
- Thoroughly tested the application for proper rendering, functionality, and Redux store updates.

### CODE REVIEW

**Comment by Kamba56** [comment]
Hi @Roland-Ntwali, @boniface193

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Your readme file is incomplete. It is missing the table of contents. Kindly add all missing sections.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/Redux/Mission/missionSlice.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/Redux/Mission/missionSlice.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/Redux/Rocket/rocketSlice.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/Redux/Rocket/rocketSlice.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/component/Missions.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchMission, spacejet } from '../Redux/Mission/missionSlice';
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/component/Missions.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/component/MyProfile.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/component/MyProfile.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/component/Nav.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/component/Nav.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/component/RocketProfile.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/component/RocketProfile.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/component/missionProfile.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/component/missionProfile.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/component/rocketDisplay.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/component/rocketDisplay.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/tests/Missions.test.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import Missions from '../component/Missions';
+import store from '../Redux/store';
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/tests/Missions.test.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/tests/MyProfile.test.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import MyProfile from '../component/MyProfile';
+import store from '../Redux/store';
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/tests/MyProfile.test.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: src/tests/Rocket.test.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import Rockets from '../component/Rocket';
+import store from '../Redux/store';
+/* eslint-disable */
```
Comment (arslanbisharat) [inline src/tests/Rocket.test.js]: - [ ] Please don't disable linters as these are here to help write clean code.

File: README.md [hunk 12]
```diff
@@ -1,70 +1,178 @@
...
 
-The page will reload when you make changes.\
-You may also see any lint errors in the console.
+- [Space Travelers Hub](#space-travelers-hub)
+- [📗 Table of Contents](#-table-of-contents)
+- [📖 \[Space Travelers Hub ](#-space-travelers-hub-)
+  - [🛠 Built With ](#-built-with-)
+    - [Tech Stack ](#tech-stack-)
+    - [Key Features ](#key-features-)
+  - [🚀 Live Demo ](#-live-demo-)
```
Comment (arslanbisharat) [inline README.md]: - [ ] Please add a proper link or remove this section to make your Readme more professional.

File: README.md [hunk 13]
```diff
@@ -1,70 +1,178 @@
...
+    - [Setup](#setup)
+    - [Install](#install)
+    - [Usage](#usage)
+    - [Run tests](#run-tests)
+  - [👥 Authors ](#-authors-)
+  - [🔭 Future Features ](#-future-features-)
+  - [🤝 Contributing ](#-contributing-)
+  - [⭐️ Show your support ](#️-show-your-support-)
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [❓ FAQ (OPTIONAL) ](#-faq-optional-)
```
Comment (arslanbisharat) [inline README.md]: - [ ] Please add a proper faqs or remove this section to make your Readme more professional.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @Roland-Ntwali, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/MyProfile.test.js (hunk 10 above)
Comment (DuaneDave) [inline src/tests/MyProfile.test.js]: As previously requested, kindly enable the linkers to check this file in order to ensure that the best practices are followed and to catch potential syntax errors.

File: src/tests/Rocket.test.js (hunk 11 above)
Comment (DuaneDave) [inline src/tests/Rocket.test.js]: As previously requested, kindly enable the linkers to check this file in order to ensure that the best practices are followed and to catch potential syntax errors.

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @team 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Excellent job resolving most of the linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @Roland-Ntwali

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by andersonlebon — APPROVED** [review]
## Hi @Roland-Ntwali

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

nodding-yes

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**